afkak/consumer.py

```diff
@@ -718,14 +718,21 @@ def _handle_fetch_error(self, failure):
         TODO: Possibly make this differentiate based on the failure
         """
         # The _request_d deferred has fired, clear it.
+        self._request_d = None
+
         if failure.check(OffsetOutOfRangeError):
-            offset_request = OffsetRequest(
-                self.topic, self.partition, OFFSET_EARLIEST, 1)
+            if self.auto_offset_reset == 'earliest':
+                offset_request = OffsetRequest(
+                    self.topic, self.partition, OFFSET_EARLIEST, 1)
+            elif self.auto_offset_reset == 'latest':
+                offset_request = OffsetRequest(
+                    self.topic, self.partition, OFFSET_LATEST, 1)
+            else:
+                self._start_d.errback(failure)
+                return
             self._request_d = self.client.send_offset_request([offset_request])
-            self._request_d.addCallbacks(
-                self._handle_offset_response, self._handle_offset_error)
+            self._request_d.addCallbacks(self._handle_offset_response, self._handle_offset_error)
```

Comment:
I see that I've used this form in a least a couple of places, but I think that this form:
https://github.com/ciena/afkak/blob/master/afkak/consumer.py#L937-L941
may be better, because if there's a coding error in _handle_offset_response, then the _handle_offset_error code can catch it.  With the addCallbacks form, I don't the error-handler "side" can handle errors in the success "side".
@twm Probably has a more well-formed opinion on that.  But note that there's no down-stream error handling on the consumer's private `self._request.d`.